### PR TITLE
Adding tests for FilterBlobs and fixing encoding issue

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Fixes https://github.com/Azure/azure-sdk-for-go/issues/17421

Golang's Encoding method returns '+' instead of '%20' for spaces, causing issues with encoding. The change is made in a transformative script. 

- [x] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/internal/generated/autorest.md [hunk 1]
```diff
@@ -321,3 +321,14 @@ directive:
...
+
+### Updating encoding URL, Golang adds '+' which disrupts encoding with service
+
+``` yaml
+directive:
+  - from: zz_service_client.go
+    where: $
+    transform: >-
+      return $.
+        replace(/req.Raw\(\).URL.RawQuery \= reqQP.Encode\(\)/, `req.Raw().URL.RawQuery = strings.Replace(reqQP.Encode(), "+", "%20", -1)`)
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/internal/generated/autorest.md:333]: I don't see any changes to the generated code.  Did you run `go generate` from the `generated` directory after making this change?  Also, shouldn't the tests be failing without the fix being applied?

**Review by gapra-msft — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Review by souravgupta-msft — APPROVED** [review]

**Comment by souravgupta-msft** [comment]
Please make sure to add an entry in the changelog.

File: sdk/storage/azblob/appendblob/client_test.go [hunk 2]
```diff
@@ -1540,8 +1540,8 @@ func (s *AppendBlobUnrecordedTestsSuite) TestCreateAppendBlobWithTags() {
 		Where: &where,
 	})
 	_require.Nil(err)
-	_require.Equal(*lResp.FilterBlobSegment.Blobs[0].Tags.BlobTagSet[0].Key, "GO ")
-	_require.Equal(*lResp.FilterBlobSegment.Blobs[0].Tags.BlobTagSet[0].Value, ".Net")
+	_require.Equal(len(lResp.FilterBlobSegment.Blobs[0].Tags.BlobTagSet), 1)
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/appendblob/client_test.go:1543]: Nit: use `_require.Len()`

**Pull request merged**